staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -306,17 +306,24 @@ func TestApfExecuteMultipleRequests(t *testing.T) {
 	defer cancel()
 	StartPriorityAndFairnessWatermarkMaintenance(ctx.Done())
 
-	var wg sync.WaitGroup
-	wg.Add(concurrentRequests)
+	errGotCh := make(chan error, concurrentRequests)
+	defer close(errGotCh)
 	for i := 0; i < concurrentRequests; i++ {
 		go func() {
```

Comment:
Wouldn't it be simpler to call t.Helper() here instead of refactoring?

Reply:
I am not sure I understand the suggestion.  https://pkg.go.dev/testing@go1.20.1#T.Helper does not say anything about modifying the rule about only calling certain functions from the main goroutine (which is what this PR is about fixing).

Reply:
I was confused because this isn't calling one of those functions (Error is OK) -- I guess it looks like this change is more about adding the timeout.

Reply:
I think placing the asserts (Error, Fatal) at the end of the test makes it more easy to follow, but for what it's worth, this test is small enough, so I reverted the changes to these tests.